clang/lib/DPCT/ApiMapping/runtime-api.inc

```diff
@@ -0,0 +1,1120 @@
+"      cudaArrayGetInfo(&desc, &extent, &flags, array);\n"
+"    }\n"
+"Is migrated to SYCL code:\n"
+"    #include <sycl/sycl.hpp>\n"
+"    #include <dpct/dpct.hpp>\n"
+"    int main() {\n"
+"      dpct::image_channel desc;\n"
+"      sycl::range<3> extent;\n"
+"      unsigned int flags;\n"
+"      dpct::image_matrix_p array;\n"
+"      dpct::image_channel channel;\n"
+"      array = new dpct::image_matrix(channel, sycl::range<2>(1, 1));\n"
+"      desc = array->get_channel();\n"
+"      extent = array->get_range();\n"
+"      flags = 0;\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaFree",
+"CUDA API:\n"
+"    cudaError_t cudaFree(void* devPtr)\n"
+"Migrates to SYCL API:\n"
+"    void sycl::free(void *ptr, const queue &q _CODELOCPARAM(&CodeLoc))\n"
+"For example, the CUDA code:\n"
+"    int main() {\n"
+"      float *p = NULL;\n"
+"      cudaMalloc((void **)&p, sizeof(p));\n"
+"      cudaFree(p);\n"
+"    }\n"
+"Is migrated to SYCL code:\n"
+"    #include <sycl/sycl.hpp>\n"
+"    #include <dpct/dpct.hpp>\n"
+"    int main() {\n"
+"      dpct::device_ext &dev_ct1 = dpct::get_current_device();\n"
+"      sycl::queue &q_ct1 = dev_ct1.default_queue();\n"
+"      float *p = NULL;\n"
+"      p = (float *)sycl::malloc_device(sizeof(p), q_ct1);\n"
+"      sycl::free(p, q_ct1);\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaFreeArray",
+"CUDA API:\n"
+"    cudaError_t cudaFreeArray(cudaArray_t array)\n"
+"Migrates to SYCL API:\n"
+"    delete\n"
+"For example, the CUDA code:\n"
+"    int main() {\n"
+"      cudaArray_t array;\n"
+"      cudaChannelFormatDesc channel;\n"
+"      cudaMallocArray(&array, &channel, 1, 1);\n"
+"      cudaFreeArray(array);\n"
+"    }\n"
+"Is migrated to SYCL code:\n"
+"    #include <sycl/sycl.hpp>\n"
+"    #include <dpct/dpct.hpp>\n"
+"    int main() {\n"
+"      dpct::image_matrix_p array;\n"
+"      dpct::image_channel channel;\n"
+"      array = new dpct::image_matrix(channel, sycl::range<2>(1, 1));\n"
+"      delete array;\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaFreeHost",
+"CUDA API:\n"
+"    cudaError_t cudaFreeHost(void* ptr)\n"
+"Migrates to SYCL API:\n"
+"    void sycl::free(void *ptr, const queue &q _CODELOCPARAM(&CodeLoc))\n"
+"For example, the CUDA code:\n"
+"    int main() {\n"
+"      float *p = NULL;\n"
+"      cudaMalloc((void **)&p, sizeof(p));\n"
+"      cudaFreeHost(p);\n"
+"    }\n"
+"Is migrated to SYCL code:\n"
+"    #include <sycl/sycl.hpp>\n"
+"    #include <dpct/dpct.hpp>\n"
+"    int main() {\n"
+"      dpct::device_ext &dev_ct1 = dpct::get_current_device();\n"
+"      sycl::queue &q_ct1 = dev_ct1.default_queue();\n"
+"      float *p = NULL;\n"
+"      p = (float *)sycl::malloc_device(sizeof(p), q_ct1);\n"
+"      sycl::free(p, q_ct1);\n"
+"    }\n"
+)
+REGISTER_ENTRY(
+"cudaGetSymbolAddress",
+"CUDA API:\n"
+"    cudaError_t cudaGetSymbolAddress(void** devPtr, const void* symbol)\n"
+"Migrates to helper function:\n"
+"    value_t *dpct::detail::device_memory::get_ptr()\n"
```

Comment:
for detail:: namespace,  it should be not used in migration code directly,  so here need update to use the publich availalbe type in dpct:: namespace.